### DISCUSSION TITLE
Get current song data

### DISCUSSION THREAD

**Discussion by TheRedM** [AzuraCast/AzuraCast#4390] [Q&A]
Hello, i'm totally noob in writing scripts, i just want a simple way to automatically get the current song in a .txt file and the content of that file changes when the song change, i already saw multiple discussions about this but non of them suits my needs.

**Comment by BusterNeece** [comment]
@TheRedM If you're looking to get that info on the same server where AzuraCast is running, we actually write a file to `/var/azuracast/stations/(station_shortcode)/config/nowplaying.txt` with the current artist and track. That file is located in the `station_data` volume, which you can access either by overriding the Docker Compose config and adding your own code into it, or by looking in the directory Docker itself uses for storage (`/var/lib/docker/volumes/azuracast_station_data/_data/`).

If you need to pull it on an external server, your best bet is to make a regular HTTP request to the Now Playing API endpoint (the first one of these should work fine: https://docs.azuracast.com/en/developers/apis/now-playing-data ) and pull the same information from the JSON that produces.

**Reply by TheRedM** [reply]
@SlvrEagle23  Thank you so much that was very helpful, i actually got it working using python :

```python
import requests
import json
azura_API = requests.get('http://xx.xx.xx.xx/api/nowplaying/1')
data = azura_API.text
parse_json = json.loads(data)
song_text = parse_json['now_playing']['song']['text']
print("Now Playing:", song_text)
with open('/**/**/song.txt','w') as fd:
    fd.write(song_text)
```

The problem now is i don't know how to keep it listening to the api for new data then rewrite it to the .txt file, i don't want to use a  cron job or a daemon because i don't want to stress the system by executing the same script every second or two
